src/shared/telemetry/clienttelemetry.d.ts

```diff
@@ -0,0 +1,104 @@
+import {Request} from 'aws-sdk/lib/request';
+import {Response} from 'aws-sdk/lib/response';
+import {AWSError} from 'aws-sdk/lib/error';
+import {Service} from 'aws-sdk/lib/service';
+import {ServiceConfigurationOptions} from 'aws-sdk/lib/service';
+import {ConfigBase as Config} from 'aws-sdk/lib/config';
+interface Blob {}
+declare class ClientTelemetry extends Service {
+  /**
+   * Constructs a service object. This object has one method for each API operation.
+   */
+  constructor(options?: ClientTelemetry.Types.ClientConfiguration)
+  config: Config & ClientTelemetry.Types.ClientConfiguration;
+  /**
+   * 
+   */
+  postErrorReport(params: ClientTelemetry.Types.PostErrorReportRequest, callback?: (err: AWSError, data: {}) => void): Request<{}, AWSError>;
+  /**
+   * 
+   */
+  postErrorReport(callback?: (err: AWSError, data: {}) => void): Request<{}, AWSError>;
+  /**
+   * 
+   */
+  postMetrics(params: ClientTelemetry.Types.PostMetricsRequest, callback?: (err: AWSError, data: {}) => void): Request<{}, AWSError>;
+  /**
+   * 
+   */
+  postMetrics(callback?: (err: AWSError, data: {}) => void): Request<{}, AWSError>;
+}
+declare namespace ClientTelemetry {
+  export type AWSProduct = "AWS Toolkit For JetBrains"|"AWS Toolkit For Eclipse"|"AWS Toolkit for VisualStudio"|string;
```

Comment:
Is there a reason to include the other products?

Reply:
Not in particular, this file is autogenerated from our service json. I'll push out a revision with the latest json and a header on this file stating that it was autogenerated.